src/odemis/gui/util/img.py

```diff
@@ -721,70 +721,64 @@ def ar_to_export_data(projections, raw=False):
 
     if raw:  # csv
-        # single image for raw ar data (phi/theta representation) for one ebeam pos
+        # single image for raw AR data (phi/theta representation) for one ebeam pos
         # if multiple images per ebeam pos (e.g. polarization or polarimetry data): batch export
         return projection.projectAsRaw()
 
     else:  # png, tiff
-        # single image for visualized ar data for one ebeam pos
+        # single image for visualized AR data for one ebeam pos
         # batch export for polarization analyzer raw data and polarimetry results for one ebeam pos
         data_dict = projection.projectAsVis()
 
         # create the image with web overlay, legend etc.
-        for pol_mode in data_dict:
-            stream_im = data_dict[pol_mode]
+        for pol_mode, stream_im in data_dict.items():
             if stream_im is None:
                 raise LookupError("Stream %s has no data selected" % (projection.name.value,))
-            plot_im = format_rgba_darray(stream_im)  # RGB -> BGRA for cairo
-            # image is always centered, fitting the whole canvas, with scale 1
-            images = set_images([(plot_im, (0, 0), (1, 1), False, None, None, None, None,
-                                  projection.name.value, None, None, {})])
-            ar_margin = int(0.2 * stream_im.shape[0])
-            ar_size = stream_im.shape[0] + ar_margin, stream_im.shape[1] + ar_margin
+
+            if stream_im.shape[0] < MIN_AR_SIZE or stream_im.shape[1] < MIN_AR_SIZE:
+                # Increase the size of the image, to fit the minimum size
+                scale = MIN_AR_SIZE / min(stream_im.shape[:2])
+                im_scale = (scale, scale)
+            else:  # Use the image as-is, and adjust the overlay/legend size to fit
+                im_scale = (1.0, 1.0)
+
+            im_size = int(round(stream_im.shape[0] * im_scale[0])), int(round(stream_im.shape[1] * im_scale[1]))
+            ar_margin = int(0.2 * im_size[0])
+            buffer_size = im_size[0] + ar_margin, im_size[1] + ar_margin
 
             # Make surface based on the maximum resolution
-            data_to_draw = numpy.zeros((ar_size[1], ar_size[0], 4), dtype=numpy.uint8)
+            data_to_draw = numpy.zeros((buffer_size[1], buffer_size[0], 4), dtype=numpy.uint8)
             surface = cairo.ImageSurface.create_for_data(
-                data_to_draw, cairo.FORMAT_ARGB32, ar_size[0], ar_size[1])
+                data_to_draw, cairo.FORMAT_ARGB32, buffer_size[0], buffer_size[1])
             ctx = cairo.Context(surface)
 
-            im = images[0]
-            buffer_center = (-ar_margin / 2, ar_margin / 2)
-            buffer_scale = (1.0, 1.0)
-            buffer_size = stream_im.shape[0], stream_im.shape[1]
-
+            # image is always centered, fitting the whole canvas, with scale 1
+            plot_im = format_rgba_darray(stream_im)  # RGB -> BGRA for cairo
+            plot_im.metadata['dc_keepalpha'] = False
```

Comment:
This tells the draw_image() function that although the image is BGRA (because that's the only format that Cairo accepts), the alpha channel shouldn't be used (because it's just been allocated and contains garbage data). Ideally draw_image() would take this as an argument, as for all the other metadata, but I didn't want to change that function at the same time.

I'll add a comment.